File: src/apis/update.js
import { invoke } from "@tauri-apps/api/core"


export async function update_check() {
    return await invoke("update_check")
}

export async function update_config_check(uconfig) {
    return await invoke("update_config_check", { uconfig })
}

export async function update_readme_check(ureadme) {
    return await invoke("update_readme_check", { ureadme })
}

File: src-tauri/src/apis/update.rs
use crate::errors::Result;
use config::update::Update;
use config::update::VerData;
use config::views::config_view::ConfigViews;
use services::update;

#[tauri::command(async)]
pub async fn update_check() -> Result<Update> {
    Ok(update::update_check().await?)
}

#[tauri::command(async)]
pub async fn update_config_check(uconfig: VerData) -> Result<ConfigViews> {
     Ok(update::config_check(&uconfig).await?)
}

#[tauri::command(async)]
pub async fn update_readme_check(ureadme: VerData) -> Result<String> {
     Ok(update::readme_check(&ureadme).await?)
}
